File: src/java/UglyNumber.java
public class UglyNumber {

    public static boolean isUglyNumber(long num) {
        if(num < 1)
            return false;

        int[] divisors = new int[]{2,3,5};
        for (int i : divisors) {
            while(num % i == 0) {
                num /= i;
            }
        }

        return num == 1;
    }

}

File: src/dart/ugly_number.dart
bool isUglyNumber(int num) {
  if (num < 1) return false;
  var divisors = [2, 3, 5];
  for (var i in divisors) {
    while (num % i == 0) num = (num / i) as int;
  }
  return num == 1;
}
